Get SchwiftyTests/CompleterTests.swift

```diff
@@ -0,0 +1,2477 @@
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "atest"
+        let expectedRange = NSMakeRange(1, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_AtTheEnd_InsertedNormally() {
+        let text = "test"
+        let range = NSMakeRange(4, 0) // End
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "testa"
+        let expectedRange = NSMakeRange(5, 0) // End
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_InTheMiddleOfWhitespace_InsertedNormally() {
+        let text = "tes\t  \tt"
+        let range = NSMakeRange(5, 0) // After the first space
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes\t a \tt"
+        let expectedRange = NSMakeRange(6, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_BetweenTabs_InsertedNormally() {
+        let text = "tes\t\tt"
+        let range = NSMakeRange(4, 0) // Between the tabs
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes\ta\tt"
+        let expectedRange = NSMakeRange(5, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_BetweenSpaces_InsertedNormally() {
+        let text = "tes  t"
+        let range = NSMakeRange(4, 0) // After the first space
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes a t"
+        let expectedRange = NSMakeRange(5, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_WithEmojis_InsertedNormally() {
+        let text = "👱🏾🚀test"
```

Comment:
@uliwitness Thank you for your thoughts on this.
So you think country flags are significantly different from emojis with a skin tone modifier? I could, of course, add some tests for country flags. The flags I tried out in the playground all had a utf16 count of 4.
As to your 3rd point: I tried to find an emoji that only has a utf16 count of 1 in the playground, but couldn't find one. Even the simplest emojis like 🙂 and 👍 seem to count at least 2.